core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -180,18 +181,52 @@ private[spark] class PythonWorkerFactory(pythonExec: String, envVars: Map[String
         return
       }
 
+      var serverSocket: ServerSocket = null
       try {
+        // get a server socket so that the launched daemon can tell us its server port
+        serverSocket = new ServerSocket(0, 0, InetAddress.getByAddress(Array(127, 0, 0, 1)))
```

Comment:
not sure it makes a difference here, but it's generally a good idea to call `setReuseAddress(true)`.

Reply:
Oh, and this one can also be a `val` instantiated outside the try.

Reply:
>but it's generally a good idea to call setReuseAddress(true).

Do I want to do this since I am asking for some available port rather than binding to a known port?